src/FileSystem-Core/AbstractFileReference.class.st

```diff
@@ -157,14 +157,14 @@ AbstractFileReference >> binaryReadStreamDo: aBlock [
 AbstractFileReference >> binaryReadStreamDo: doBlock ifAbsent: absentBlock [
 	^ self isFile 
 		ifTrue: [ self binaryReadStreamDo: doBlock ]
-		ifFalse: absentBlock
+		ifFalse: [ absentBlock value ]
```

Comment:
Should not it be part of compiler optimization logic? 
It's strange that we are forced to write such code.

Reply:
Yes, in the long term this is the goal. There was a first experiment (see https://github.com/pharo-project/pharo/pull/8791).

As soon as we solve this better, we of course will remove the rules.